craft.lic

```diff
@@ -0,0 +1,291 @@
+      train_forging
+    end
+
+    if args.craft == 'alchemy'
+      exit if DRSkill.getxp('Alchemy') > @craft_max_mindstate
+      train_alchemy
+    end
+  end
+
+  def train_outfitting
+    if @settings.train_workorders.include?('Tailoring')
+      return unless money_for_training?(5000, 'Outfitting')
+      wait_for_script_to_complete('workorders', ['Tailoring'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    rank = DRSkill.getrank('Outfitting')
+    if rank <= 25 # Tier 1  Extremely Easy
+      sew(5, 'knitted socks', 'socks')
+    elsif rank <= 50 # Tier 2 Very Easy
+      sew(5, 'knitted mittens', 'mittens')
+    elsif rank <= 100 # Tier 3  Easy
+      sew(5, 'knitted hat', 'hat')
+    elsif rank <= 175 # Tier 4  Simple
+      sew(5, 'knitted gloves', 'gloves')
+    elsif rank <= 300 # Tier 5  Basic
+      sew(5, 'knitted hose', 'hose')
+    elsif rank <= 425 # Tier 6  Somewhat Challenging
+      sew(5, 'knitted cloak', 'cloak')
+    elsif rank <= 650 # Tier 7  Challenging
+      sew(5, 'knitted blanket', 'blanket')
+    else # Tier 12  Extremely Difficult
+      echo('*** NOT YET IMPLEMENTED ***')
+    end
+  end
+
+  def train_engineering
+    if @settings.train_workorders.include?('Shaping')
+      return unless money_for_training?(5000, 'Engineering')
+      wait_for_script_to_complete('workorders', ['Shaping'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    elsif @settings.train_workorders.include?('Carving')
+      return unless money_for_training?(5000, 'Engineering')
+      wait_for_script_to_complete('workorders', ['Carving'])
+      wait_for_script_to_complete('sell-loot')
+      walk_to(@training_room)
+      return
+    end
+
+    rank = DRSkill.getrank('Engineering')
+
+    if rank <= 25 # Tier 1  Extremely Easy
+      shape(7, 'wood band', 'band')
+    elsif rank <= 50 # Tier 2 Very Easy
+      shape(7, 'wood bracelet', 'bracelet')
+    elsif rank <= 100 # Tier 3  Easy
+      shape(7, 'wood cloak pin', 'cloak pin')
+    elsif rank <= 175 # Tier 4  Simple
+      shape(7, 'wood pendant', 'pendant')
+    elsif rank <= 300 # Tier 5  Basic
+      shape(7, 'wood brooch', 'brooch')
+    elsif rank <= 425 # Tier 6  Somewhat Challenging
+      shape(7, 'wood armband', 'armband')
+    elsif rank <= 600 # Tier 7  Challenging
+      shape(7, 'wood choker', 'choker')
+    elsif rank <= 700 # Tier 8 - Complicated
+      shape(7, 'articulated wood necklace', 'necklace')
+    elsif rank <= 850 # Tier 9 - Intricate
+      shape(7, 'wood crown', 'crown')
+    elsif rank <= 1175 # Tier 10 - Difficult
+      shape(7, 'wood comb', 'comb')
+    elsif rank <= 1400 # Tier 11 - Very Difficult
+      shape(7, 'wood haircomb', 'haircomb')
+    else # Tier 12  Extremely Difficult
+      echo('*** NOT YET IMPLEMENTED ***')
+    end
+  end
+
+  def train_forging
+    eligible = @settings.train_workorders & %w[Blacksmithing Weaponsmithing]
```

Comment:
💡 Should engineering (shaping/carving) be refactored to use this pattern?

Reply:
good call. also needed in present day crossing-training

Reply:
deleted the workorders handling from this completely based on later comments, it will stay in crossing-training

Reply:
I have a commit ready to make this change after the money change to crossing-training goes in.